src/Data/Text/Show.hs

```diff
@@ -46,18 +54,65 @@ unpack ::
 unpack = S.unstreamList . stream
 {-# INLINE [1] unpack #-}
 
--- | /O(n)/ Convert a literal string into a 'Text'.
---
--- This is exposed solely for people writing GHC rewrite rules.
+-- | /O(n)/ Convert a null-terminated
+-- <https://en.wikipedia.org/wiki/UTF-8#Modified_UTF-8 modified UTF-8>
```

Comment:
There seems to be a difference from Modified UTF-8. Aren't 4-byte UTF-8 codepoints still represented as themselves (i.e., in 4 bytes) in Haskell literals (which is why we can afford to erase surrogates), whereas Java's Modified UTF-8 actually must encode them as surrogate pairs?

Reply:
Yes, Haskell version of Modified UTF-8 represent 4-byte codepoints transparently. "**All known** Modified UTF-8 implementations also treat the surrogate pairs as in CESU-8" is false, Wiki should say "Most".